nengo/params.py

```diff
@@ -187,18 +233,37 @@ def __set__(self, instance, ndarray):
             ndarray = self.validate(instance, ndarray)
         self.data[instance] = ndarray
 
-    def validate(self, instance, ndarray):
-        ndim = len(self.shape)
-        try:
-            ndarray = np.asarray(ndarray, dtype=np.float64)
-        except TypeError:
-            raise ValueError("Must be a float NumPy array (got type '%s')"
-                             % ndarray.__class__.__name__)
-
-        if ndarray.ndim != ndim:
+    def validate(self, instance, ndarray):  # noqa: C901
+        if isinstance(ndarray, np.ndarray):
+            ndarray = ndarray.view()
+        else:
+            try:
+                ndarray = np.array(ndarray, dtype=np.float64)
+            except TypeError:
+                raise ValueError("Must be a float NumPy array (got type '%s')"
+                                 % ndarray.__class__.__name__)
+        if self.readonly:
+            ndarray.setflags(write=False)
+
+        if '...' in self.shape:
```

Comment:
Maybe a comment here would be helpful... something like `# Convert '...' to the appropriate number of '*' dimensions` (if I'm reading this correctly)